### ISSUE TITLE
Struct.$clone() must clone nested structs

### ISSUE THREAD

**Issue by AndreyZM** [bridgedotnet/Bridge#1804]
### Steps To Reproduce
http://deck.net/75a54e5c5a0632cbd4ea9b08332136b1

```csharp
public struct Struct1
{
    public Struct2 nested;
}

public struct Struct2
{
    public int field;
}

public class Program
{
    static public void Main()
    {
        var a = new Struct1();
        var b = a;
        a.nested.field = 5;
        Console.WriteLine("Expected 5:" + a.nested.field);//Actual 5
        Console.WriteLine("Expected 0:" + b.nested.field);//Actual 5
    }
}
```
### Actual
```js
$clone: function (to) {
            var s = to || new Demo.Struct2();
            s.field = this.field;
            return s;
        }
```

### Expected
```js
$clone: function (to) {
            var s = to || new Demo.Struct2();
            this.field.clone(s.field);
            return s;
        }
```

**Comment by geoffreymcgill** [comment]
@AndreyZM - Thanks for the simplified sample. This really helps us isolate and reproduce issue. 

Here's a DotNetFiddle demonstrating the same scenario:
https://dotnetfiddle.net/G9XOA9

We will do out best to fix this right away and get into the 15.0 release.

**Comment by LeonidVeriga** [comment]
Related to #1613
